common/dynamicconfig/constants.go

```diff
@@ -1327,6 +1327,12 @@ checks while a shard is lingering.`,
 will temporarily delay closing shards after a membership update, awaiting a
 shard ownership lost error from persistence. If set to zero, shards will not delay closing.
 Do NOT use non-zero value with persistence layers that are missing AssertShardOwnership support.`,
+	)
+	ShardFinalizerTimeLimit = NewGlobalDurationSetting(
+		"history.shardFinalizerTimeLimit",
+		2*time.Second,
```

Comment:
Q: Is this a reasonable timeout?

Reply:
How many updates can be aborted within 2 seconds? I bet quite a lot. Let's start with 2 seconds and see. I don't have better number.

Reply:
No clue, really. Let's start there 👍